### DISCUSSION TITLE
Documentation des mécanismes

### DISCUSSION THREAD

**Discussion by simonparisot** [publicodes/publicodes#61] [General]
Bonjour,
Nous cherchons à utiliser publicodes (grammaire et engine) dans un projet financier et il me semble que les mécanismes manquent cruellement de documentation :)
Nous avons cherché par exemple à comprendre le fonctionnement simple de allégement dans le cas d'une décote, et n'avons trouvé aucune documentation associée. 
Finalement le mécanisme est simple et via les règles et exemples déjà rédigés et à l'aide du studio (https://publi.codes/studio) il est possible d'intuiter le fonctionnement. Mais une documentation des principaux mécanismes me parait indispensable à une réutilisation par des tiers.

**Comment by laem** [comment]
J'ai vu avec plaisir apparaître en ligne un début de documentation des mécanismes ici : https://publi.codes/mécanismes

**Comment by rchampourlier** [comment]
@simonparisot avez-vous trouvé une réponse à votre question ? Je m'intéresse à ce moteur pour un projet personnel et je cherche à voir à quel point il peut répondre à mon besoin.

Un peu déçu de ne pas voir le code source sur ce repo, mais par contre il est dispo [ici](https://github.com/betagouv/mon-entreprise/blob/master/publicodes) 😉

**Comment by lajarre** [comment]
Bonjour @rchampourlier , la documentation de Publicodes est dispo sur https://publi.codes/ avec en particulier la documentation sur les mécanismes: https://publi.codes/documentation/m%C3%A9canismes

Par ailleurs le package Publicodes est en effet pour l'instant en monorepo avec celui de mon-entreprise ainsi que les règles associées (package `modele-social`). Ceci est sans doute amené à changer à l'avenir :)

Enfin, pour continuer la discussion: https://gitter.im/publicodes

**Comment by rchampourlier** [comment]
Bonjour @lajarre,

Merci beaucoup pour la réponse et les détails. J'ai raté la partie mécanismes à mon passage sur la doc 🤔, désolé !

Hâte d'en voir plus (je vais lire le code pour m'initier un peu !) et merci pour le gitter :)
